Packet++/header/HttpLayer.h

```diff
@@ -501,7 +524,15 @@ namespace pcpp
 		 * But the user can set a non-default status code string and it will be written in the header first line. Empty string ("") means using the
 		 * default status code string
 		 */
-		HttpResponseLayer(HttpVersion version, HttpResponseStatusCode statusCode, std::string statusCodeString = "");
+		PCPP_DEPRECATED explicit HttpResponseLayer(HttpVersion version, const HttpResponseStatusCode& statusCode, const std::string& statusCodeString);
```

Comment:
since we already support `HttpResponseStatusCode` constructor to set customized message, this constructor should be deprecated.

Reply:
I think we should keep the original method signature and add back the default value: 
```cpp
PCPP_DEPRECATED HttpResponseLayer(HttpVersion version, HttpResponseStatusCode statusCode, std::string statusCodeString = "");
```

Also - let's add to the doxygen comment that it's deprecated and users should use the other constructor?

Reply:
it will conflict with the new API
`explicit HttpResponseLayer(HttpVersion version, const HttpResponseStatusCode& statusCode);`

from the user's perspective, they will only get the deprecated message if they use three parameters before. If they just use two parameters before, there is no difference for them

Reply:
Yes you're right, sorry my bad